src/DssSpell.t.sol

```diff
@@ -11,21 +11,70 @@ interface Gem6Like {
 
 interface GemJoin6Like {
     function implementations(address) external view returns (uint256);
+    function join(address, uint256) external;
+    function exit(address, uint256) external;
+}
+
+interface AuthLike {
+    function wards(address) external view returns (uint256);
 }
 
 contract DssSpellTest is DssSpellTestBase {
 
-    function testTUSDImplementation() public {
+    function test_TUSDImplementation() public {
+        address TUSD = addr.addr("TUSD");
+        address MCD_JOIN_TUSD_A = addr.addr("MCD_JOIN_TUSD_A");
+        address prevImpl = 0xffc40F39806F1400d8278BfD33823705b5a4c196;
+        address nextImpl = 0xd8D59c59Ab40B880b54C969920E8d9172182Ad7b;
+        address actualImpl = Gem6Like(TUSD).implementation();
+        assertEq(nextImpl, actualImpl);
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(prevImpl), 1);
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(nextImpl), 0);
+
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "DssSpellTest/spell-not-done");
+
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(prevImpl), 0);
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(nextImpl), 1);
+    }
+
+    function joinAndExitTUSD() internal {
         address TUSD = addr.addr("TUSD");
         address MCD_JOIN_TUSD_A = addr.addr("MCD_JOIN_TUSD_A");
-        address implementation = Gem6Like(TUSD).implementation();
-        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(implementation), 0);
+        address ali = 0x6C64CeAF2338cE509A612cf244FFb121e0ff172c;
```

Comment:
What is this address? lol. It appears to be in use. This test will fail if whoever this is has an existing balance of TUSD. I would just transfer to and back from the current address. No need for a third party.

Reply:
done